From bdd7aeddc3e7b870ce8971f3277e9a49182bc8e5 Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Tue, 14 Sep 2010 08:30:14 +0000
Subject: [PATCH 0268/1092] cbus: tahvo: remove device_release

it's useless as we don't do anything besides
wait_for_completion(). Remove it.

Signed-off-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: Tony Lindgren <tony@atomide.com>
---
 drivers/cbus/tahvo.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/drivers/cbus/tahvo.c b/drivers/cbus/tahvo.c
index 7d49252..7c57ca4 100644
--- a/drivers/cbus/tahvo.c
+++ b/drivers/cbus/tahvo.c
@@ -56,8 +56,6 @@ static int tahvo_is_betty;
 static struct tasklet_struct tahvo_tasklet;
 spinlock_t tahvo_lock = SPIN_LOCK_UNLOCKED;
 
-static struct completion device_release;
-
 struct tahvo_irq_handler_desc {
 	int (*func)(unsigned long);
 	unsigned long arg;
@@ -388,11 +386,6 @@ static int __devexit tahvo_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static void tahvo_device_release(struct device *dev)
-{
-	complete(&device_release);
-}
-
 static struct platform_driver tahvo_driver = {
 	.probe		= tahvo_probe,
 	.remove		= __devexit_p(tahvo_remove),
@@ -404,9 +397,6 @@ static struct platform_driver tahvo_driver = {
 static struct platform_device tahvo_device = {
 	.name		= "tahvo",
 	.id		= -1,
-	.dev = {
-		.release = tahvo_device_release,
-	}
 };
 
 /**
@@ -422,8 +412,6 @@ static int __init tahvo_init(void)
 		machine_is_nokia_n810() || machine_is_nokia_n810_wimax()))
 			return -ENODEV;
 
-	init_completion(&device_release);
-
 	ret = platform_driver_register(&tahvo_driver);
 	if (ret)
 		return ret;
@@ -443,7 +431,6 @@ static void __exit tahvo_exit(void)
 {
 	platform_device_unregister(&tahvo_device);
 	platform_driver_unregister(&tahvo_driver);
-	wait_for_completion(&device_release);
 }
 
 subsys_initcall(tahvo_init);
-- 
1.7.11.2

